packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -249,393 +314,657 @@ describe("Decentraland", () => {
+            const functionName = "approve";
+            const argsArray = [MARKETPLACE_ADDRESS, manaAmountForShopping];
+            const action = gnosisSafe.customContractAction(
+              contractAddress,
+              contractABI,
+              functionName,
+              argsArray
+            );
+
+            const confirmationListener = async message => {
+              const owner = GNOSIS_SAFE_ADDRESS;
+              const spender = MARKETPLACE_ADDRESS;
+              const allowance = await mana.allowance(owner, spender);
+              expect(`${allowance}`).to.equal(`${manaAmountForShopping}`);
+
+              done();
+            };
+
+            const errorListener = error => {
+              action.off("error");
+              done(error);
+            };
+
+            action.once("confirmation", confirmationListener);
+            action.on("error", errorListener);
+
+            action.send();
+          })();
+        });
 
-          await executeOrderAction.send();
-          await executeOrderAction.waitForOneConfirmation();
+        it("should buy an estate", done => {
+          (async () => {
+            // Global hooks don't run between same level hooks
+            await mineBlocks(provider, 1);
+
+            const {
+              assetId,
+              nftAddress,
+              seller,
+              priceInWei,
+              expiresAt,
+            } = estateForSale;
+
+            const buyerAddress = GNOSIS_SAFE_ADDRESS;
+            await checkAsset(estate, mana, estateForSale, buyerAddress);
+
+            await expectExactTokenBalances(estate, [GNOSIS_SAFE_ADDRESS], [0]);
+
+            // Gnosis Safe should execute an open order
+            // TODO: ephemeralWallet should broadcast this action
+            const contractAddress = marketplace.getAddress();
+            const contractABI = marketplace.getABI();
+            const functionName = "safeExecuteOrder";
+            const fingerprint = await estate.getFingerprint(`${assetId}`);
+            const argsArray = [
+              nftAddress,
+              `${assetId}`,
+              `${priceInWei}`,
+              `${fingerprint}`,
+            ];
+
+            gnosisSafe.setSigners([gnosisSafeOwner]);
+            gnosisSafe.setWallet(gnosisSafeOwner);
+            const action = gnosisSafe.customContractAction(
+              contractAddress,
+              contractABI,
+              functionName,
+              argsArray
+            );
+
+            const onFailed = message => {
+              const { data } = message;
+              const { args } = data;
+              const { txHash } = args;
+              done(new Error(`ExecutionFailed event emitted`));
+            };
+
+            const confirmationListener = async message => {
+              // WIP: Not working because of gas issue on Marketplace.safeExecuteOrder() call
+              //await expectExactTokenBalances(estate, [GNOSIS_SAFE_ADDRESS], [1]);
```

Comment:
So in other words, we want to be expecting that it's 1, but instead we have to expect that it's 0 until we figure out the gas thing? If so, we can say that even more explicitly in the comments

Reply:
Yes. Ok, I'll be more explicitly.

Reply:
Great, thanks!

Reply:
Done on https://github.com/tasitlabs/TasitSDK/pull/383

Reply:
Thanks! Resolving this